lib_vehicle_model/src/lib_vehicle_model/ConstraintChecker.cpp

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (C) 2018-2019 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#include <stdlib.h> 
+#include "ConstraintChecker.h"
+
+
+
+/**
+ * Cpp containing the implementation of ConstraintChecker
+ */
+using namespace lib_vehicle_model;
+
+ConstraintChecker::ConstraintChecker(std::shared_ptr<ParameterServer> parameter_server) {
+
+  // Load Parameters
+  bool speedParam = parameter_server->getParam("max_forward_speed", max_forward_speed_); 
+  bool accelParam = parameter_server->getParam("forward_acceleration_limit", forward_acceleration_limit_);
+  bool decelParam = parameter_server->getParam("forward_deceleration_limit", forward_deceleration_limit_);
+  bool maxSteerParam = parameter_server->getParam("max_steering_angle", max_steering_angle_);
+  bool minSteerParam = parameter_server->getParam("min_steering_angle", min_steering_angle_);
+  bool maxSteerRateParam = parameter_server->getParam("max_steering_angle_rate", max_steering_angle_rate_);
+  bool maxTrailerAngleParam = parameter_server->getParam("max_trailer_angle", max_trailer_angle_);
+  bool minTrailerAngleParam = parameter_server->getParam("min_trailer_angle", min_trailer_angle_);
+
+  // Check if all the required parameters could be loaded
+  if (!(speedParam && accelParam && decelParam && maxSteerParam && minSteerParam
+         && maxSteerRateParam && maxTrailerAngleParam && minTrailerAngleParam)) {
+
+    std::ostringstream msg;
```

Comment:
with std::string I would need to wrap all my input parameters in std::toString() calls. Also if I ever want to add an object to this message then I will need to add a toString function to that object instead of the << operator overload which supports all stream types. This is because std::toString cannot be overloaded like << can. 

using std::string directly might be somewhat faster (not sure of how concatenating this many strings impacts it though), but for an exception message like this I don't think that is a concern.

Reply:
Thanks for explanation. Because my initial impression for std::ostringstream is about read/write a document instead of manipulate strings. But it is clear to me now for this use case.